docs/apps/docker.md

```diff
@@ -82,73 +91,271 @@ This tutorial also provides valuable troubleshooting insights and advice on what
 
-``` Shell
+Containers are confined in a sandbox, so they can't access the host
+in any ways unless you explicitly prescribe acceptable accesses
+to the host.
+The Docker Engine provides mounting filesystems into the container with the `-v` argument and the following syntax:
+`-v some/path/in/host:/absolute/path/within/container:ro`,
+where the trailing `:ro` specifies that the mount is read-only.
+The mount permissions modifiers can be omitted, which means the mount
+will have read-write permissions.
+
+!!! warning "*Docker for Windows* requires enabling Shared Drives"
+
+    On *Windows* installations, the `-v` argument will not work
+    by default because it is necessary to enable shared drives.
+    Please check on this [Stackoverflow post](https://stackoverflow.com/a/51822083) how to enable them.
+
+In general, you'll want to at least provide two mount-points:
+one set in read-only mode for the input data and one read/write
+to store the outputs:
+
+``` {.shell hl_lines="2 3"}
+$ docker run -ti --rm \
+    -v path/to/data:/data:ro \        # read-only, for data
+    -v path/to/output:/out \          # read-write, for outputs
+    nipreps/fmriprep:<latest-version> \
+    /data /out/out \
+    participant
+```
+
+When **debugging** or **reusing pre-cached intermediate results**,
+you'll also need to mount some working directory that otherwise
+is not exposed by the application.
+In the case of *NiPreps*, we typically inform the *BIDS Apps*
+to override the work directory by setting the `-w`/`--work-dir`
+argument (please note that this is not defined by the *BIDS Apps*
+specifications and it may change across applications):
+
+``` {.shell hl_lines="4 8"}
+$ docker run -ti --rm \
+    -v path/to/data:/data:ro \
+    -v path/to/output:/out \
+    -v path/to/work:/work \              # mount from host
+    nipreps/fmriprep:<latest-version> \
+    /data /out/out \
+    participant
+    -w /work                             # override default directory
+```
+
+!!! tip "Best practices"
+
+    The [*ReproNim* initiative](https://www.repronim.org/)
+    distributes materials and documentation of best practices
+    for containerized execution of neuroimaging workflows.
+    Most of these are organized within the
+    [*YODA* (Yoda's Organigram on Data Analysis)](https://github.com/myyoda) principles.
+
+    For example, mounting `$PWD` into `$PWD` and setting that path
+    as current working directory can effectively resolve many issues.
+    This strategy may be combined with the above suggestion about
+    the application's work directory as follows:
+
+    ``` {.shell hl_lines="4 5 9"}
+    $ docker run -ti --rm \
+        -v path/to/data:/data:ro \
+        -v path/to/output:/out \
+        -v $PWD:$PWD \
+        -w $PWD \   # DO NOT confuse with the application's work directory
+        nipreps/fmriprep:<latest-version> \
+        /data /out/out \
+        participant
+        -w $PWD/work
+    ```
```

Comment:
I would personally do this:

```suggestion
    ``` {.shell hl_lines="4 5 9"}
    $ docker run -ti --rm \
        -v $PWD:$PWD \  # Mount the current directory with its own name
        -w $PWD \       # DO NOT confuse with the application's work directory
        nipreps/fmriprep:<latest-version> \
        inputs/raw-data outputs/fmriprep \  # With YODA, the inputs are inside the working directory
        participant
        -w $PWD/work
    ```
```

(Actually, I would use `sourcedata/raw .`, but that would require additional explanation.)